scripts/updateSelectorEngine.ts

```diff
@@ -0,0 +1,105 @@
+import cp from 'node:child_process';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+/**
+ * This script updates the JQuery selector engine for the mock-doc package
+ * to the latest version.
+ *
+ * To run it, use the following command:
+ * ```sh
+ * npm run tsc.scripts
+ * npm run build.updateSelectorEngine
+ * ```
+ */
+
+const rootDir = path.resolve(__dirname, '..', '..');
+const jqueryDepDir = path.resolve(rootDir, 'node_modules', 'jquery');
+const WINDOW_MOCK = `{
+  document: {
+    createElement() {
+      return {};
+    },
+    nodeType: 9,
+    documentElement: {
+      nodeType: 1,
+      nodeName: 'HTML'
+    }
+  }
+}`;
+
+async function run() {
+  console.log('updating JQuery Sel!ector engine...');
+
+  await runCommand(`npm install --ignore-scripts`, jqueryDepDir);
+  await runCommand(`npm run build -- --include=selector`, jqueryDepDir);
+
+  const jqueryPkgJSON = JSON.parse(await fs.readFile(path.resolve(jqueryDepDir, 'package.json'), 'utf8'));
+  const thirdPartyDir = path.resolve(__dirname, '../../src/mock-doc/third-party');
+  await fs.mkdir(thirdPartyDir, { recursive: true });
+
+  const originalContent = await fs.readFile(path.resolve(jqueryDepDir, 'dist', 'jquery.js'), 'utf8');
+
+  /**
+   * This is a hack to make the jQuery selector engine work with the mock-doc package.
+   * Using the original jQuery bundle would cause a "RegExpCompiler Allocation failed - process out of memory"
+   * error due to the way the jQuery object is constructed. The following tweaks to the jQuery bundle
+   * resolve the issue:
+   */
+  const fixedJQuery = originalContent
```

Comment:
Changes like this give me hesitation, but as long as the plan is to never really change which version of jQuery we pull in, then I think we'll be okay.

Reply:
This is mostly to avoid the memory allocation issue I faced as there are tons of custom created Regular Expressions in there. This is reported in Chromium so chances are that eventually we can remove this and Node.js can handle this better. Given our Q1 roadmap it is more likely though that we nudge users to test on a different infrastructure other than our `mock-doc`, e.g. in a real headless browser, which also will remove the need for this dependency in general.